pyirf/tests/test_gammapy_interop.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+import astropy.units as u
+import numpy as np
+
+pytest.importorskip('gammapy')
+
+
+true_energy_bins = [0.1, 1, 10, 100] * u.TeV
+fov_offset_bins = [0, 1, 2] * u.deg
+source_offset_bins = np.linspace(0, 1, 20) * u.deg
+migration_bins = np.geomspace(0.2, 5, 10)
+
+
+def test_effective_area_table_2d():
+    from pyirf.gammapy import create_effective_area_table_2d
+
+    shape = (len(true_energy_bins) - 1, len(fov_offset_bins) - 1)
+    aeff = np.random.uniform(0, 1e5, size=shape) * u.m**2
+
+    aeff_gammapy = create_effective_area_table_2d(aeff, true_energy_bins, fov_offset_bins)
+    # test str repr works
```

Comment:
not sure I get what this is supposed to do...

wouldn't be more robust to write a fits file HDU and read it back like the following

`aeff = EffectiveAreaTable2D.read(filename, hdu="AEFF")`

(the scope of this comment is for all the 3 test functions)

Reply:
Why would you write a file to go from one memory-representation to another? What would be more robust about introducing a completely unnececessary round-trip through the fits format instead of just using the documented gammapy API?

Reply:
They test that the conversion works. It does not assert anything, since the error case is an exception raised by gammapy (E.g. a ValueError if the axis have wrong names  or the data has wrong shape).